HHS/Head-Start-TTADP
# [Prod] Additional updates to TR form, fix bug with trailing space on URL validation

## Description of change

An error in the way we validated URLs is fixed - a trailing spaces on an objective URL resource meant that it would be marked invalid and disappear on users. This fixes that by trimming the string before validating it and sending it to the backend, meaning that a trimmed string will be return and the user will never notice the disappearing string.

Otherwise, this PR includes updates spurred on mainly through discussions with the UX team about the previous round of updates and fixes to the TR form and the session form.

## How to test

On an AR, enter an otherwise valid URL with a space at the end and hit "save goal." You will see the URL persist as you might expect.


## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1863

## Checklists

### Every PR



### Production Deploy


### After merge/deploy
